mitodl/ol-data-platform
# Create combined enrollment mart and upstream changes

# What are the relevant tickets?




https://github.com/mitodl/ol-data-platform/issues/885

# Description (What does it do?)

- Creates `marts__combined_course_enrollment_detail` with enrollments, certificates, fulfilled/refunded orders, coupons and cybersource payment details from MITx, xPro and Bootcamps. It contains the following fields
    platform 
    courserunenrollment_is_active
    courserunenrollment_created_on
    courserunenrollment_enrollment_mode
    courserunenrollment_enrollment_status
    user_id
    courserun_id
    courserun_title
    courserun_readable_id
    user_username
    user_email
    user_full_name
    courseruncertificate_created_on
    courseruncertificate_url
    order_state
    order_reference_number
    order_total_price_paid
    order_created_on
    line_price
    coupon_code
    coupon_redeemed_on
    coupon_discount_amount
    payment_transaction_id
    payment_authorization_code
    payment_method
    payment_bill_to_address_state
    payment_bill_to_address_country
- Creates `int__micromasters__orders` for legacy DEDP orders
- A few upstream changes


# How can this be tested?

dbt build --select +int__bootcamps__ecommerce_receipt 
dbt build --select +int__bootcamps__ecommerce_order 
dbt build --select +int__micromasters__orders 
dbt build --select +int__mitxonline__ecommerce_transaction 
dbt build --select +int__mitxonline__ecommerce_order
dbt build --select +int__mitxpro__ecommerce_couponpaymentversion 
dbt build --select +int__mitxpro__ecommerce_coupon
dbt build --select +int__mitxpro__ecommerce_receipt 
dbt build --select +int__mitxpro__ecommerce_order 
dbt build --select +int__mitxpro__ecommerce_line 
dbt build --select +int__mitxpro__ecommerce_allcoupons 
dbt build --select marts__combined_course_enrollment_detail

or 

just run `dbt build --select +marts__combined_course_enrollment_detail` its going to take a while

Note if you see warning about something like `dbt_expectations_expect_table_row_count_to_equal_other_table_int__mitxpro__ecommerce_productversion`, that's because the above commands don't run the sibiling of these intermediate models,  you can just run that warned intermediate model, the test should pass
